compiler/src/iree/compiler/Dialect/Flow/IR/FlowOps.cpp

```diff
@@ -879,20 +866,35 @@ FunctionType DispatchOp::getEntryPointType() {
   return FunctionType::get(getContext(), argTypes, getResultTypes());
 }
 
+std::pair<unsigned, unsigned> DispatchOp::getTiedOperandsIndexAndLength() {
+  return getODSOperandIndexAndLength(1);  // $operands
+}
+
 LogicalResult DispatchOp::verify() {
   Operation *op = getOperation();
-  if (workgroup_count().empty()) {
-    return op->emitOpError() << "at least one workgroup dimension is required";
-  }
   if (failed(verifyOpDynamicDims(op, operands(), operand_dims())) ||
       failed(verifyOpDynamicDims(op, results(), result_dims()))) {
     return failure();
   }
   return success();
 }
 
-std::pair<unsigned, unsigned> DispatchOp::getTiedOperandsIndexAndLength() {
-  return getODSOperandIndexAndLength(1);  // $operands
+LogicalResult DispatchOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+  Operation *op = getOperation();
+  auto entryPointOp = symbolTable.lookupNearestSymbolFrom(op, entry_point());
+  if (!entryPointOp) {
+    // TODO(benvanik): there are a lot of tests that are assuming this is not
+    // verified. We'll need to go add dummy executables for all of them. Today
+    // we just bail on the verifier if the symbol isn't found.
+    //
+    // Should be:
+    //   return op->emitOpError() << "undefined entry point: " << entry_point();
+    return success();
```

Comment:
Aw, this is a bit of a shame. I've found it inconvenient in tests to add supporting IR not directly related to what I want to test. More verification does sound useful though.

Sounds like updating those tests should be pretty mechanical. Could file an issue and just chip away at it.

Reply:
Yeah I agree; my thought was that because no valid programs generated by our compiler in internal dialects will have missing symbols this is effectively a mechanism for externing things. We could add external executables as a proper thing, but then you get all the boilerplate with none of the verification x_x
Thankfully the number of things that could break that would cause the symbols to be missing are like 3 lines of rarely changed code and all the hairy hard to debug stuff is going to verify below!

Reply:
In other cases, IR like `"test.sink"`, `"test.consumer1"(%arg0) : (i1) -> ()`, etc. have helped focus tests without involving lots of boilerplate. Maybe we could do something similar for executables?